src/Npgsql/ConnectorPool.cs

```diff
@@ -98,6 +99,9 @@ public override string ToString()
 
         internal ConnectorPool(NpgsqlConnectionStringBuilder settings, string connString)
         {
+            Debug.Assert(PoolSizeLimit <= short.MaxValue,
```

Comment:
Just a thought - we could make this a runtime check - there's no perf impact whatsoever (just a setup check), and really confused users may accidentally set this number very high. It's true that they'll like get errors from the PostgreSQL about surpassing max_connections, but for correctness why not do it anyway.

Reply:
I don't get what you mean with runtime check? The assert is on an internal const.